database/migrations/default/1742417174000_crash_ems_link/up.sql

```diff
@@ -0,0 +1,391 @@
+add constraint matched_crash_pks_non_null check (
+    (crash_match_status = 'multiple_matches_by_automation')
+    or (matched_crash_pks is NULL)
+);
+comment on column ems__incidents.crash_match_status is 'The status of the crash record match';
+comment on column ems__incidents.matched_crash_pks is 'The IDs of multiple crashes that were found to match this record. Can only be populated when crash_match_status is ''multiple_matches_by_automation''';
+
+--
+-- add column to assign injury severity
+--
+alter table ems__incidents
+add column patient_injry_sev_id integer generated always as (
+    case
+        when lower(pcr_provider_impression_primary) = 'death on scene' then 4
+        when pcr_outcome = 'DECEASED ON SCENE' then 4
+        when
+            pcr_patient_acuity_final
+            = 'DEAD WITHOUT RESUSCITATION EFFORTS (BLACK)'
+            then 4
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_final) = 'lower acuity (green)'
+            then 2
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_final) = 'critical (red)'
+            then 1
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_final) = 'emergent (yellow)'
+            then 1
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_initial) = 'critical (red)'
+            then 1
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_initial) = 'emergent (yellow)'
+            then 1
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_level) = 'high acuity'
+            then 1
+        -- what is this pcr_transport_priority?
+        when
+            upper(left(pcr_transport_priority, 1)) = 'CHARLIE'
+            or upper(left(pcr_transport_priority, 1)) = 'D'
+            then 1
+        when
+            upper(right(pcr_transport_priority, 1)) = '3'
+            and lower(pcr_patient_acuity_initial) = 'lower acuity (green)'
+            then 2
+        when
+            upper(left(pcr_transport_priority, 1)) = 'minor injury'
+            or upper(left(pcr_transport_priority, 1)) = 'possible injury'
+            then 2
+        when
+            upper(right(pcr_transport_priority, 1)) != '3'
+            and lower(pcr_patient_acuity_initial) = 'lower acuity (green)'
+            then 3
+        when
+            upper(right(pcr_outcome, 7)) = 'REFUSED'
+            and left(lower(pcr_patient_acuity_level), 10) = 'low acuity'
+            then 3
+        else 5
+    end
+) stored;
+
+alter table ems__incidents
+add constraint ems__incidents_patient_injry_sev_id_fk foreign key (
+    patient_injry_sev_id
+)
+references lookups.injry_sev on update restrict on delete restrict;
```

Comment:
so if this lookup tables ids were ever updated this constraint would prevent that update?

Reply:
yes. although it seems very unlikely we would ever do that, i got a little paranoid that we would forget that this non-CRIS table is also dependent on the same lookup table as `people` 🤷